src/components/sbb-card/sbb-card.tsx

```diff
@@ -0,0 +1,50 @@
+import { Component, h, Host, JSX, Prop, State } from '@stencil/core';
+import { InterfaceSbbCardAttributes } from './sbb-card.custom';
+
+/**
+ * @slot unnamed - Slot to render the content.
+ * @slot badge - Slot to render `<sbb-card-badge>`.
+ */
+
+@Component({
+  shadow: true,
+  styleUrl: 'sbb-card.scss',
+  tag: 'sbb-card',
+})
+export class SbbCard {
+  /**
+   * Size variant, either xs, s, m, l, xl and xxl.
+   */
+  @Prop() public size?: InterfaceSbbCardAttributes['size'] = 'm';
+
+  /* @internal */
+  @State() private _hasBadge = false;
+
+  /**
+   * It is used internally to show the `<sbb-card>`.
+   *
+   * @param size The size selected.
+   * @returns True whether size is equal to m, l, xl or xxl.
+   */
+  private _showSBBBadge(size: string): boolean {
+    return size === 'm' || size === 'l' || size === 'xl' || size === 'xxl';
+  }
+
+  public render(): JSX.Element {
+    return (
+      <Host class={{ card__badge: this._showSBBBadge(this.size) && this._hasBadge }}>
```

Comment:
Why is the "this._hasBadge" check necessary? If the slot has no content, it would not be displayed anyways.

Reply:
Removing the check means that the host takes a class even if no slot content is provided, resulting in a wrong padding.

Reply:
Ok, I see, thank you.